fix(nix): wrangler 4.93.0 source-build fails on darwin (EBADF), breaking rust-worker devshells

## Problem

The rust-worker nix devshell fails to enter on macOS (aarch64-darwin)
because `pkgs.wrangler@4.93.0` has **no binary substitute** in
cache.nixos.org, so nix builds it from source — and that source build
**deterministically fails** on darwin in wrangler's tsup/DTS step:

```
DTS ⚡️ Build success in 6703ms
DTS wrangler-dist/cli.d.ts 125.68 KB
AggregateError: EBADF: bad file descriptor, fstat
    at FSReqCallback.readFileAfterClose (node:internal/fs/read/context:52:21)
ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  wrangler@4.93.0 build:
  `pnpm run clean && pnpm tsup && pnpm run generate-json-schema`
```

Failing derivation: `wrangler-4.93.0.drv`
(`/nix/store/i2fm55rsqyqfqiin12zwdm4wklnq5g25-wrangler-4.93.0.drv`).
It fails on every attempt, not flakily.

### Impact

- Any rust-worker consumer's devshell is unenterable on darwin: the
  whole `nix-shell-env` fails on the wrangler dependency, so
  `shaka`/`cargo`/`rustc`/`worker-build` never land on PATH.
- First hit in **buzzingo** (`apps/buzzingo/flake.nix`, nixpkgs pinned
  `bd0ff2d3eac2`), which blocks all local work there.
- **CI (Linux) is unaffected** — wrangler builds/substitutes fine on
  ubuntu — so `main` stays green while local darwin shells are broken.
  This is a darwin-only, source-build-only failure.

The EBADF-after-"DTS Build success" signature is the known tsup/pnpm
fd race under the nix sandbox; on darwin there's no cached output to
fall back to, so it surfaces every time.

## Proposed implementation

Stop the devshell from source-building wrangler on darwin. Options,
roughly in order of preference:

- Pin wrangler to a version that has a darwin binary substitute in
  cache.nixos.org (avoid source build entirely), or vendor a prebuilt
  wrangler (e.g. npm-bin / fetchurl of the published package) instead
  of the source-built derivation.
- Make wrangler a *non-blocking* / optional devshell input on darwin —
  it's only needed at deploy time (CI), not for local Rust dev — so a
  broken wrangler build doesn't gate shell entry.
- If keeping the source build, patch the derivation to dodge the EBADF
  (serialize the tsup CJS/DTS passes, or split the DTS step) — fragile,
  last resort.

Whichever lands should flow through the rust-worker flake template
(see #831) so consumers get the fix on regenerate.

## Acceptance criteria

- [ ] `nix develop` / direnv `use flake` enters cleanly on
      aarch64-darwin for a rust-worker project (buzzingo as the test
      case).
- [ ] `shaka`, `cargo`, `rustc`, `worker-build` resolve on PATH in the
      shell on darwin.
- [ ] No regression on Linux/CI (wrangler still available for deploy).
- [ ] Fix is reflected in the rust-worker flake template, not just
      hand-patched in one consumer.

## Out of scope

- Buzzingo's hand-authored `flake.nix` is still un-generated pending
  #831's external-consumer support; fully replacing it is tracked
  there, not here.

Related: #831 (generate flake.nix for rust-worker), #333 (rust-worker
devshell deploy ergonomics).

--- a/apps/cascadiajs2026-notes/flake.nix
+++ b/apps/cascadiajs2026-notes/flake.nix
@@ -57,7 +57,6 @@
           default = pkgs.mkShell {
             packages = [
               (rustToolchain pkgs)
-              pkgs.wrangler
               pkgs.worker-build
               pkgs.pkg-config
               pkgs.openssl
@@ -65,6 +64,7 @@
               pkgs.cue
             ]
             ++ (workflowPackages pkgs)
+            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler
             ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;
 
             shellHook = ''

--- a/apps/kolohelios-portfolio/flake.nix
+++ b/apps/kolohelios-portfolio/flake.nix
@@ -57,7 +57,6 @@
           default = pkgs.mkShell {
             packages = [
               (rustToolchain pkgs)
-              pkgs.wrangler
               pkgs.worker-build
               pkgs.pkg-config
               pkgs.openssl
@@ -65,6 +64,7 @@
               pkgs.cue
             ]
             ++ (workflowPackages pkgs)
+            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler
             ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;
 
             shellHook = ''

--- a/apps/notes-web/flake.nix
+++ b/apps/notes-web/flake.nix
@@ -57,12 +57,12 @@
           default = pkgs.mkShell {
             packages = [
               (rustToolchain pkgs)
-              pkgs.wrangler
               pkgs.worker-build
               pkgs.pkg-config
               pkgs.openssl
             ]
             ++ (workflowPackages pkgs)
+            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler
             ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;
 
             shellHook = ''

--- a/apps/pollen-alert/flake.nix
+++ b/apps/pollen-alert/flake.nix
@@ -57,12 +57,12 @@
           default = pkgs.mkShell {
             packages = [
               (rustToolchain pkgs)
-              pkgs.wrangler
               pkgs.worker-build
               pkgs.pkg-config
               pkgs.openssl
             ]
             ++ (workflowPackages pkgs)
+            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler
             ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;
 
             shellHook = ''

--- a/services/notes-sync/flake.nix
+++ b/services/notes-sync/flake.nix
@@ -57,12 +57,12 @@
           default = pkgs.mkShell {
             packages = [
               (rustToolchain pkgs)
-              pkgs.wrangler
               pkgs.worker-build
               pkgs.pkg-config
               pkgs.openssl
             ]
             ++ (workflowPackages pkgs)
+            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler
             ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;
 
             shellHook = ''

--- a/tools/shaka/src/project/generate_flakes.rs
+++ b/tools/shaka/src/project/generate_flakes.rs
@@ -728,13 +728,22 @@ const WORKER_OUTPUT_HEADER: &str = r#"    {
 "#;
 
 /// Devshell block for worker flakes. Fixed base — the worker toolchain,
-/// `wrangler`, `worker-build`, `pkg-config`, `openssl` — then any
+/// `worker-build`, `pkg-config`, `openssl` — then any
 /// `extra_dev_shell_packages`, then `workflowPackages` and the Linux-only
-/// `cargo-llvm-cov`, closing with the `~/.cargo/bin` PATH `shellHook`
-/// (appended, not prepended, so a runner's rustup `cargo` doesn't shadow
-/// nix's toolchain). `worker-build` is pinned via nixpkgs rather than
-/// `cargo install`ed at runtime so its wasm-bindgen minimum can't drift by
-/// runner cache (#864).
+/// `wrangler` and `cargo-llvm-cov`, closing with the `~/.cargo/bin` PATH
+/// `shellHook` (appended, not prepended, so a runner's rustup `cargo`
+/// doesn't shadow nix's toolchain). `worker-build` is pinned via nixpkgs
+/// rather than `cargo install`ed at runtime so its wasm-bindgen minimum
+/// can't drift by runner cache (#864).
+///
+/// `wrangler` is gated to Linux (`lib.optional isLinux`) because its
+/// nixpkgs derivation has no darwin binary substitute and source-builds
+/// deterministically fail on darwin (EBADF in tsup's DTS step), which
+/// would gate the whole shell — `shaka`/`cargo`/`worker-build` never reach
+/// PATH. It's only needed at deploy time, and `cf-deploy.yml` runs on
+/// `ubuntu-latest` via `nix develop . --command`, so it must stay in the
+/// **default** shell on Linux (a separate deploy shell breaks CI verify).
+/// See #877.
 ///
 /// With `consumes_shaka`, the devShell function destructures `system`
 /// (needed to index `shaka.packages.${system}`) and the `shellHook`
@@ -754,7 +763,6 @@ fn render_worker_devshell(extra_dev_shell_packages: &[String], consumes_shaka: b
     out.push_str("          default = pkgs.mkShell {\n");
     out.push_str("            packages = [\n");
     out.push_str("              (rustToolchain pkgs)\n");
-    out.push_str("              pkgs.wrangler\n");
     out.push_str("              pkgs.worker-build\n");
     out.push_str("              pkgs.pkg-config\n");
     out.push_str("              pkgs.openssl\n");
@@ -763,6 +771,9 @@ fn render_worker_devshell(extra_dev_shell_packages: &[String], consumes_shaka: b
     }
     out.push_str("            ]\n");
     out.push_str("            ++ (workflowPackages pkgs)\n");
+    out.push_str(
+        "            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler\n",
+    );
     out.push_str(
         "            ++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.cargo-llvm-cov;\n",
     );
@@ -1227,6 +1238,18 @@ mod tests {
         assert!(out.contains("pkgs.pkg-config"));
         assert!(out.contains("pkgs.openssl"));
         assert!(out.contains(r#"export PATH="$PATH:$HOME/.cargo/bin""#));
+    }
+
+    #[test]
+    fn worker_wrangler_is_linux_gated() {
+        // wrangler source-builds (and deterministically fails) on darwin,
+        // which would gate the whole shell. It's gated to Linux so darwin
+        // shells enter, while cf-deploy.yml (ubuntu) still gets it from the
+        // default shell. See #877.
+        let out = render_worker_flake("pollen-alert", &WorkerMeta::default());
+        assert!(out.contains("++ pkgs.lib.optional pkgs.stdenv.hostPlatform.isLinux pkgs.wrangler"));
+        // Not in the unconditional package list.
+        assert!(!out.contains("              pkgs.wrangler\n"));
     }
 
     #[test]

